package.json

```diff
@@ -125,6 +125,7 @@
     "@elastic/apm-rum": "^5.2.0",
     "@elastic/charts": "19.5.2",
     "@elastic/datemath": "5.0.3",
+    "@elastic/elasticsearch": "^7.7.1",
```

Comment:
The client follows the stack versioning, meaning that using the client `7.x` in kibana `master` will cause issues. You should use the client `master` branch.
Currently, we are not publishing any `8.x` version on npm, but we could do it if it does help you.
[Here](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/introduction.html#_compatibility) you can find the compatibility table of the client.
If you want to install the master branch of the client:
```
npm install elastic/elasticsearch-js#master
```

Reply:
Is the client released separately for **every** Stack release? Should it be another place to sync across the Stack when bumping the Kibana version?

Reply:
Yes, we do a release for every `major.minor` of the stack, patches are released as soon as it's needed.

Reply:
Hum, this may be quite problematic AFAIK, as kibana master is targeting 8.0, but the current branch (7.9 atm for example) is target 7.9

That means that we would need to have different version (with potential differences in APIs) between our kibana master and our next-release branch?

This feels like it could become a backport nightmare, doesn't it?

Reply:
@delvedor Can you elaborate on the typical changes between versions? If newer versions only change to support or remove new or deprecated ES functionality then this shouldn't cause any problems for us that aren't already caused by ES.

But if elasticsearch-js plans to make breaking changes to it's API signatures this adds an additional maintenance burden and we will have to migrate all code within the release timeframe.

Reply:
The client follows semantic versioning, so there will never be a breaking change between minor or patch releases, but there might be between majors.
Minor releases are always additive, in a generic minor release you will find new ES endpoints and some additional features of the client, for example, in the last 2/3 minors, client helpers have been added.

If the client needs to do a breaking change, which can be dropping the support for a specific version of Node, remove/change a configuration option, or drop an API, that will happen in a major release.

The only parts of the client that could have a breaking change between minors are the helpers and the type definitions, which are still experimental (even if they are stable and not expected to change unless there is a very good reason).

Reply:
The client does prereleases as soon as there is a feature freeze, if you take a look at the [published versions](https://www.npmjs.com/package/@elastic/elasticsearch) on npm you will see few rcs.